stdlib/public/Python/Python.swift

```diff
@@ -1253,6 +1253,21 @@ extension PythonObject : MutableCollection {
   }
 }
 
+extension PythonObject : IteratorProtocol, Sequence {
```

Comment:
I recommend against the technique of making sequences their own iterators too. It's usually a better choice to declare a nested `Iterator` type, ensuring that the sequence is unambiguously multi-pass with a fresh iterator from the start on each call to `makeIterator()`.

Reply:
I am not sure I get this part correctly

Reply:
You could make a nested Iterator struct which would wrap a PythonObject and just expose next(). This way, it is more type-safe if someone tries to iterate a python sequence themselves.

Reply:
You can declare a ‘struct Iterator : IteratorProtocol {}’ within a PythonObject extension.

Reply:
I updated the PR. Is this internal struct okay to be public here?